### PR TITLE
Fix internal data type

### PR DESCRIPTION
**Pull request by MykolaVasyk**
# Title

Fix error of internal data type

## Motivation

Improving App Store Connect Client

## Modifications

Added BuildsQuery.swift file

## Result

It is planned to correct the error that occurred during the integration of the App Store Connect Client

### CODE REVIEW

File: Sources/AppStoreConnectClient/AppStoreConnectClient.swift [hunk 1]
```diff
@@ -125,17 +125,24 @@ public struct AppStoreConnectClient {
     public func fetchBuilds(
         for app: Application,
         with query: BuildsQuery
-    ) async throws -> [Build] {
+    ) async throws -> [Build]? {
+        guard let querySort = query.sort else {
```
Comment (mcmikius) [inline Sources/AppStoreConnectClient/AppStoreConnectClient.swift]: sort

File: Sources/AppStoreConnectClient/AppStoreConnectClient.swift [hunk 2]
```diff
@@ -125,17 +125,24 @@ public struct AppStoreConnectClient {
...
+                errors: "\(String(describing: query.sort))"
+            )
+        }
+        guard let sort = Sort(rawValue: querySort.rawValue) else {
+            throw AppStoreConnectError.invalidSortParameter(
+                errors: "\(String(describing: Sort(rawValue: querySort.rawValue)))"
+            )
+        }
+        let sortPayload = query.convertSort(sort)
+        let fieldsPayload = query.convertFields([])
```
Comment (mcmikius) [inline Sources/AppStoreConnectClient/AppStoreConnectClient.swift]: It is not correct 'cause you have empty array

File: Sources/AppStoreConnectClient/AppStoreConnectClient.swift (hunk 1 above)
Comment (mcmikius) [reply Sources/AppStoreConnectClient/AppStoreConnectClient.swift]: sort may be empty

File: Sources/AppStoreConnectClient/AppStoreConnectError.swift [hunk 3]
```diff
@@ -38,4 +38,8 @@ public enum AppStoreConnectError: Error, Equatable {
     case notFound(errors: String?)
     /// An unexpected error occurred.
     case unexpectedError(errors: String?)
+    /// A required sort parameter is missing from the request.
+    case missingSortParameter(errors: String?)
```
Comment (mcmikius) [inline Sources/AppStoreConnectClient/AppStoreConnectError.swift]: This error no need

File: Sources/AppStoreConnectClient/Models/BuildsQuery.swift [hunk 4]
```diff
@@ -27,7 +27,74 @@ import Foundation
...
-    let sort: Operations.builds_hyphen_get_collection.Input.Query.sortPayload
+    public let sort: Sort?
     /// The fields payload for the query.
-    let fields: Operations.builds_hyphen_get_collection.Input.Query.fields_lbrack_builds_rbrack_Payload
+    public let fields: [Fields]?
+    
+    /// Converts the Sort enum to the appropriate payload type used in the query.
+    /// - Parameter from: The Sort enum value to convert.
+    /// - Returns: The converted sort payload.
+    func convertSort(_ from: Sort) -> Operations.builds_hyphen_get_collection.Input.Query.sortPayloadPayload {
```
Comment (mcmikius) [inline Sources/AppStoreConnectClient/Models/BuildsQuery.swift]: I think it is incorrect 'cause sort is array

File: Sources/AppStoreConnectClient/Models/BuildsQuery.swift [hunk 5]
```diff
@@ -27,7 +27,74 @@ import Foundation
...
+    /// - Returns: An array of converted field payloads.
+    func convertFields(_ from: [Fields]) -> [Operations.builds_hyphen_get_collection.Input.Query.fields_lbrack_builds_rbrack_PayloadPayload] {
+        return from.compactMap { field in
+            return Operations.builds_hyphen_get_collection.Input.Query.fields_lbrack_builds_rbrack_PayloadPayload(rawValue: field.rawValue)
+        }
+    }
+}
+
+/// An enumeration representing the possible sort options for the query.
+public enum Sort: String {
```
Comment (mcmikius) [inline Sources/AppStoreConnectClient/Models/BuildsQuery.swift]: Move to different file or nested type of BuildsQuery

File: Sources/AppStoreConnectClient/Models/BuildsQuery.swift [hunk 6]
```diff
@@ -27,7 +27,74 @@ import Foundation
...
+    case preReleaseVersion
+    case hyphenPreReleaseVersion = "-preReleaseVersion"
+    case uploadedDate
+    case hyphenUploadedDate = "-uploadedDate"
+    case version
+    case hyphenVersion = "-version"
+}
+
+/// An enumeration representing the possible fields that can be included in the query.
+public enum Fields: String {
```
Comment (mcmikius) [inline Sources/AppStoreConnectClient/Models/BuildsQuery.swift]: Move to different file or nested type of BuildsQuery

File: Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift [hunk 7]
```diff
@@ -200,14 +200,14 @@ final class AppStoreConnectClientTests: XCTestCase {
...
-                fields: .init(arrayLiteral: .version, .minOsVersion, .uploadedDate)
+                sort: .hyphenVersion,
+                fields: [.version, .minOsVersion, .uploadedDate]
             )
         )
-        XCTAssertEqual(build.first?.id, expectedBuilds.first?.id)
-        XCTAssertEqual(build.first?.version, expectedBuilds.first?.version)
-        XCTAssertEqual(build.first?.uploadedDate, expectedBuilds.first?.uploadedDate)
-        XCTAssertEqual(build.first?.minOsVersion, expectedBuilds.first?.minOsVersion)
+        XCTAssertEqual(build?.first?.id, expectedBuilds.first?.id)
```
Comment (mcmikius) [inline Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift]: Result must be comparable and then compare object

**Review by mcmikius — COMMENTED** [review]
Need resolve comments

File: Sources/AppStoreConnectClient/AppStoreConnectClient.swift (hunk 1 above)
Comment (MykolaVasyk) [reply Sources/AppStoreConnectClient/AppStoreConnectClient.swift]: Сorrected.

File: Sources/AppStoreConnectClient/AppStoreConnectClient.swift (hunk 2 above)
Comment (MykolaVasyk) [reply Sources/AppStoreConnectClient/AppStoreConnectClient.swift]: Changed.

File: Sources/AppStoreConnectClient/AppStoreConnectError.swift (hunk 3 above)
Comment (MykolaVasyk) [reply Sources/AppStoreConnectClient/AppStoreConnectError.swift]: Deleted.

File: Sources/AppStoreConnectClient/Models/BuildsQuery.swift (hunk 6 above)
Comment (MykolaVasyk) [reply Sources/AppStoreConnectClient/Models/BuildsQuery.swift]: replaced to BuildsQuery.

File: Sources/AppStoreConnectClient/Models/BuildsQuery.swift (hunk 5 above)
Comment (MykolaVasyk) [reply Sources/AppStoreConnectClient/Models/BuildsQuery.swift]: Replaced to BuildsQuery.

File: Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift (hunk 7 above)
Comment (MykolaVasyk) [reply Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift]: Changed to XCTAssertEqual(build, expectedBuilds)

File: Sources/AppStoreConnectClient/Models/BuildsQuery.swift (hunk 4 above)
Comment (MykolaVasyk) [reply Sources/AppStoreConnectClient/Models/BuildsQuery.swift]: Changed.

File: Sources/AppStoreConnectClient/AppStoreConnectClient.swift [hunk 8]
```diff
@@ -125,17 +125,14 @@ public struct AppStoreConnectClient {
     public func fetchBuilds(
         for app: Application,
         with query: BuildsQuery
-    ) async throws -> [Build] {
+    ) async throws -> [Build]? {
```
Comment (mcmikius) [inline Sources/AppStoreConnectClient/AppStoreConnectClient.swift]: Why return parameter became optional?

File: Sources/AppStoreConnectClient/AppStoreConnectClient.swift [hunk 9]
```diff
@@ -125,17 +125,14 @@ public struct AppStoreConnectClient {
...
+        let sortPayload = query.convertSort(query.sort ?? .init( arrayLiteral: .hyphenVersion))
```
Comment (mcmikius) [inline Sources/AppStoreConnectClient/AppStoreConnectClient.swift]: It is incorrect

Comment (mcmikius) [reply Sources/AppStoreConnectClient/AppStoreConnectClient.swift]: I think this must be empty array

Comment (mcmikius) [reply Sources/AppStoreConnectClient/AppStoreConnectClient.swift]: Incorrect name of the constant

File: Sources/AppStoreConnectClient/AppStoreConnectClient.swift [hunk 10]
```diff
@@ -125,17 +125,14 @@ public struct AppStoreConnectClient {
...
+        let fieldsPayload = query.convertFields(query.fields ?? [])
```
Comment (mcmikius) [inline Sources/AppStoreConnectClient/AppStoreConnectClient.swift]: Incorrect name of the constant

File: Sources/AppStoreConnectClient/AppStoreConnectClient.swift [hunk 11]
```diff
@@ -125,17 +125,14 @@ public struct AppStoreConnectClient {
     public func fetchBuilds(
         for app: Application,
         with query: BuildsQuery
```
Comment (mcmikius) [inline Sources/AppStoreConnectClient/AppStoreConnectClient.swift]: I think it is must be optional

File: Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift [hunk 12]
```diff
@@ -220,11 +217,11 @@ final class AppStoreConnectClientTests: XCTestCase {
             let result = try await client.fetchBuilds(
                 for: app,
                 with: BuildsQuery.init(
-                    sort: .init(arrayLiteral: ._hyphen_version),
-                    fields: .init(arrayLiteral: .version, .minOsVersion, .uploadedDate)
+                    sort: .init(arrayLiteral: .hyphenVersion),
```
Comment (mcmikius) [inline Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift]: must be array

File: Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift [hunk 13]
```diff
@@ -242,11 +239,11 @@ final class AppStoreConnectClientTests: XCTestCase {
...
```
Comment (mcmikius) [inline Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift]: must be array

File: Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift [hunk 14]
```diff
@@ -264,11 +261,11 @@ final class AppStoreConnectClientTests: XCTestCase {
...
```
Comment (mcmikius) [inline Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift]: must be array

File: Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift [hunk 15]
```diff
@@ -286,11 +283,11 @@ final class AppStoreConnectClientTests: XCTestCase {
...
```
Comment (mcmikius) [inline Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift]: must be array

**Review by mcmikius — COMMENTED** [review]
Need resolve comments

File: Sources/AppStoreConnectClient/AppStoreConnectClient.swift (hunk 8 above)
Comment (MykolaVasyk) [reply Sources/AppStoreConnectClient/AppStoreConnectClient.swift]: Changed.

File: Sources/AppStoreConnectClient/AppStoreConnectClient.swift (hunk 9 above)
Comment (MykolaVasyk) [reply Sources/AppStoreConnectClient/AppStoreConnectClient.swift]: Changed to "sort".

File: Sources/AppStoreConnectClient/AppStoreConnectClient.swift (hunk 10 above)
Comment (MykolaVasyk) [reply Sources/AppStoreConnectClient/AppStoreConnectClient.swift]: Changed to "fields".

File: Sources/AppStoreConnectClient/AppStoreConnectClient.swift (hunk 11 above)
Comment (MykolaVasyk) [reply Sources/AppStoreConnectClient/AppStoreConnectClient.swift]: Made optional.

File: Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift (hunk 12 above)
Comment (MykolaVasyk) [reply Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift]: Changed.

File: Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift (hunk 13 above)
Comment (MykolaVasyk) [reply Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift]: Changed.

File: Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift (hunk 14 above)
Comment (MykolaVasyk) [reply Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift]: Changed.

File: Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift (hunk 15 above)
Comment (MykolaVasyk) [reply Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift]: Changed.

File: Sources/AppStoreConnectClient/AppStoreConnectClient.swift [hunk 16]
```diff
@@ -124,18 +124,15 @@ public struct AppStoreConnectClient {
     /// - Throws: An error of type `AppStoreConnectError` if the response indicates an error.
     public func fetchBuilds(
         for app: Application,
-        with query: BuildsQuery
+        with query: BuildsQuery?
     ) async throws -> [Build] {
+        let sort = query?.convertSort(query?.sort ?? .init( arrayLiteral: .hyphenVersion))
```
Comment (mcmikius) [inline Sources/AppStoreConnectClient/AppStoreConnectClient.swift]: It is incorrect. This parameter should not have a default value.

**Review by mcmikius — COMMENTED** [review]
Need resolve comments

File: Sources/AppStoreConnectClient/AppStoreConnectClient.swift (hunk 16 above)
Comment (MykolaVasyk) [reply Sources/AppStoreConnectClient/AppStoreConnectClient.swift]: Corrected.

File: Sources/AppStoreConnectClient/Models/BuildsQuery.swift [hunk 17]
```diff
@@ -27,7 +27,63 @@ import Foundation
...
-    let sort: Operations.builds_hyphen_get_collection.Input.Query.sortPayload
+    public let sort: [Sort]?
     /// The fields payload for the query.
-    let fields: Operations.builds_hyphen_get_collection.Input.Query.fields_lbrack_builds_rbrack_Payload
+    public let fields: [Fields]?
+    
+    /// Converts the Sort enum to the appropriate payload type used in the query.
+    /// - Parameter from: The Sort enum value to convert.
+    /// - Returns: The converted sort payload.
+    func convertSort(_ from: [Sort]?) -> [Operations.builds_hyphen_get_collection.Input.Query.sortPayloadPayload]? {
```
Comment (mcmikius) [inline Sources/AppStoreConnectClient/Models/BuildsQuery.swift]: `func convertSort(_ from: [Sort]?) -> [Operations.builds_hyphen_get_collection.Input.Query.sortPayloadPayload]?` it is so-so need something like this `func convertSort(from sorts: [Sort]) -> [Operations.builds_hyphen_get_collection.Input.Query.sortPayloadPayload]`

File: Sources/AppStoreConnectClient/Models/BuildsQuery.swift [hunk 18]
```diff
@@ -27,7 +27,63 @@ import Foundation
...
+        return from?.compactMap { sort in
+            return Operations.builds_hyphen_get_collection.Input.Query.sortPayloadPayload(rawValue: sort.rawValue)
+        }
+    }
+    
+    /// Converts the Fields enum array to the appropriate payload type used in the query.
+    /// - Parameter from: The array of Fields enum values to convert.
+    /// - Returns: An array of converted field payloads.
+    func convertFields(_ from: [Fields]?) -> [Operations.builds_hyphen_get_collection.Input.Query.fields_lbrack_builds_rbrack_PayloadPayload]? {
```
Comment (mcmikius) [inline Sources/AppStoreConnectClient/Models/BuildsQuery.swift]: `func convertFields(_ from: [Fields]?) -> [Operations.builds_hyphen_get_collection.Input.Query.fields_lbrack_builds_rbrack_PayloadPayload]?`
it is so-so decision. Need something like this `func convert(from fields: [Fields]) -> [Operations.builds_hyphen_get_collection.Input.Query.fields_lbrack_builds_rbrack_PayloadPayload]`

File: Sources/AppStoreConnectClient/Models/BuildsQuery.swift [hunk 19]
```diff
@@ -27,7 +27,63 @@ import Foundation
 /// A structure representing a query for builds with specified sort and fields parameters.
 public struct BuildsQuery {
     /// The sort payload for the query.
-    let sort: Operations.builds_hyphen_get_collection.Input.Query.sortPayload
+    public let sort: [Sort]?
```
Comment (mcmikius) [inline Sources/AppStoreConnectClient/Models/BuildsQuery.swift]: I think the best solution use empty array as default value. Something like this - `public var sort: [Sort] = []`

File: Sources/AppStoreConnectClient/Models/BuildsQuery.swift [hunk 20]
```diff
@@ -27,7 +27,63 @@ import Foundation
...
     /// The fields payload for the query.
-    let fields: Operations.builds_hyphen_get_collection.Input.Query.fields_lbrack_builds_rbrack_Payload
+    public let fields: [Fields]?
```
Comment (mcmikius) [inline Sources/AppStoreConnectClient/Models/BuildsQuery.swift]: I think the best solution use empty array as default value. Something like this - `public var fields: [Fields] = []`

File: Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift [hunk 21]
```diff
@@ -200,14 +200,11 @@ final class AppStoreConnectClientTests: XCTestCase {
         let build = try await client.fetchBuilds(
             for: app,
             with: BuildsQuery.init(
```
Comment (mcmikius) [inline Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift]: Need use BuildsQuery(...

File: Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift [hunk 22]
```diff
@@ -220,11 +217,11 @@ final class AppStoreConnectClientTests: XCTestCase {
             let result = try await client.fetchBuilds(
                 for: app,
                 with: BuildsQuery.init(
```
Comment (mcmikius) [inline Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift]: Need use BuildsQuery(...

File: Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift [hunk 23]
```diff
@@ -242,11 +239,11 @@ final class AppStoreConnectClientTests: XCTestCase {
...
```
Comment (mcmikius) [inline Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift]: Need use BuildsQuery(...

File: Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift [hunk 24]
```diff
@@ -264,11 +261,11 @@ final class AppStoreConnectClientTests: XCTestCase {
...
```
Comment (mcmikius) [inline Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift]: Need use BuildsQuery(...

File: Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift [hunk 25]
```diff
@@ -286,11 +283,11 @@ final class AppStoreConnectClientTests: XCTestCase {
...
```
Comment (mcmikius) [inline Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift]: Need use BuildsQuery(...

File: Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift [hunk 26]
```diff
@@ -389,4 +386,36 @@ final class AppStoreConnectClientTests: XCTestCase {
...
+    
+    func testFetchBuildsSuccessWithSortIsNil() async throws {
+        var mockClient = MockAPIClient()
+        mockClient.result = .ok
+        let client = AppStoreConnectClient(client: mockClient)
+        let app = Application(id: "Foo", bundleId: "Bar")
+        let expectedBuilds = [Build(schema: MockObjects.build)]
+        let build = try await client.fetchBuilds(
+            for: app,
+            with: BuildsQuery.init(
```
Comment (mcmikius) [inline Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift]: Need use BuildsQuery(...

File: Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift [hunk 27]
```diff
@@ -389,4 +386,36 @@ final class AppStoreConnectClientTests: XCTestCase {
...
+    
+    func testFetchBuildsSuccessWithFieldsIsNil() async throws {
+        var mockClient = MockAPIClient()
+        mockClient.result = .ok
+        let client = AppStoreConnectClient(client: mockClient)
+        let app = Application(id: "Foo", bundleId: "Bar")
+        let expectedBuilds = [Build(schema: MockObjects.build)]
+        let build = try await client.fetchBuilds(
+            for: app,
+            with: BuildsQuery.init(
```
Comment (mcmikius) [inline Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift]: Need use BuildsQuery(...

File: Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift [hunk 28]
```diff
@@ -389,4 +386,36 @@ final class AppStoreConnectClientTests: XCTestCase {
...
+                sort: [.hyphenVersion, .uploadedDate],
+                fields: nil
+            )
+        )
+        XCTAssertEqual(build, expectedBuilds)
+    }
```
Comment (mcmikius) [inline Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift:420]: Need add test without query. SUT - `let build = try await client.fetchBuilds(for: app)`

**Review by mcmikius — COMMENTED** [review]
Need resolve comments

File: Sources/AppStoreConnectClient/Models/BuildsQuery.swift (hunk 17 above)
Comment (MykolaVasyk) [reply Sources/AppStoreConnectClient/Models/BuildsQuery.swift]: Changed.

File: Sources/AppStoreConnectClient/Models/BuildsQuery.swift (hunk 18 above)
Comment (MykolaVasyk) [reply Sources/AppStoreConnectClient/Models/BuildsQuery.swift]: Changed.

File: Sources/AppStoreConnectClient/Models/BuildsQuery.swift (hunk 19 above)
Comment (MykolaVasyk) [reply Sources/AppStoreConnectClient/Models/BuildsQuery.swift]: Changed.

File: Sources/AppStoreConnectClient/Models/BuildsQuery.swift (hunk 20 above)
Comment (MykolaVasyk) [reply Sources/AppStoreConnectClient/Models/BuildsQuery.swift]: Changed.

File: Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift (hunk 21 above)
Comment (MykolaVasyk) [reply Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift]: Changed.

File: Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift (hunk 22 above)
Comment (MykolaVasyk) [reply Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift]: Changed.

File: Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift (hunk 23 above)
Comment (MykolaVasyk) [reply Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift]: Changed.

File: Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift (hunk 24 above)
Comment (MykolaVasyk) [reply Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift]: Changed.

File: Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift (hunk 25 above)
Comment (MykolaVasyk) [reply Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift]: Changed.

File: Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift (hunk 26 above)
Comment (MykolaVasyk) [reply Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift]: Changed.

File: Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift (hunk 27 above)
Comment (MykolaVasyk) [reply Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift]: Changed.

File: Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift (hunk 28 above)
Comment (MykolaVasyk) [reply Tests/AppStoreConnectClientTests/AppStoreConnectClientTests.swift:420]: Fixed.

File: Sources/AppStoreConnectClient/AppStoreConnectClientProtocol.swift [hunk 29]
```diff
@@ -57,5 +57,5 @@ public protocol AppStoreConnectClientProtocol {
     /// - Returns: a `PreReleaseVersion` object containing details about the retrieved pre-release version.
     /// - Throws:AppStoreConnectClientError: An error that may occur during the fetch process.
     /// - Note: Pre-release versions are used for beta testing or/and internal reviews before an app is published on the App Store.
-    func fetchPreReleaseVersion(by id: Build) async throws -> PreReleaseVersion
+    func fetchPreReleaseVersion(for id: Build) async throws -> PreReleaseVersion
```
Comment (mcmikius) [inline Sources/AppStoreConnectClient/AppStoreConnectClientProtocol.swift]: (by id:

File: Sources/AppStoreConnectClient/Models/BuildsQuery.swift [hunk 30]
```diff
@@ -27,7 +27,63 @@ import Foundation
...
-    let sort: Operations.builds_hyphen_get_collection.Input.Query.sortPayload
+    public var sort: [Sort] = []
     /// The fields payload for the query.
-    let fields: Operations.builds_hyphen_get_collection.Input.Query.fields_lbrack_builds_rbrack_Payload
+    public var fields: [Fields] = []
+    
+    /// Converts the Sort enum to the appropriate payload type used in the query.
+    /// - Parameter from: The Sort enum value to convert.
+    /// - Returns: The converted sort payload.
+    func convertSort(from sorts: [Sort]) -> [Operations.builds_hyphen_get_collection.Input.Query.sortPayloadPayload] {
```
Comment (mcmikius) [inline Sources/AppStoreConnectClient/Models/BuildsQuery.swift]: convert(from sorts: [Sort])

File: Sources/AppStoreConnectClient/Models/BuildsQuery.swift [hunk 31]
```diff
@@ -27,7 +27,63 @@ import Foundation
...
+        return sorts.compactMap { sort in
+            return Operations.builds_hyphen_get_collection.Input.Query.sortPayloadPayload(rawValue: sort.rawValue)
+        }
+    }
+    
+    /// Converts the Fields enum array to the appropriate payload type used in the query.
+    /// - Parameter from: The array of Fields enum values to convert.
+    /// - Returns: An array of converted field payloads.
+    func convertFields(from fields: [Fields]) -> [Operations.builds_hyphen_get_collection.Input.Query.fields_lbrack_builds_rbrack_PayloadPayload] {
```
Comment (mcmikius) [inline Sources/AppStoreConnectClient/Models/BuildsQuery.swift]: func convert(from fields: [Fields])

File: Sources/AppStoreConnectClient/AppStoreConnectClient.swift [hunk 32]
```diff
@@ -124,20 +124,26 @@ public struct AppStoreConnectClient {
...
-        let response = try await client.builds_hyphen_get_collection(
-            query: .init(
-                filter_lbrack_app_rbrack_: [app.id],
-                sort: type(of: query.sort).init(arrayLiteral: ._hyphen_version),
-                fields_lbrack_builds_rbrack_: type(of: query.fields).init(
-                    arrayLiteral: 
-                    .version,
-                    .minOsVersion,
-                    .uploadedDate
+        let response: Operations.builds_hyphen_get_collection.Output
```
Comment (mcmikius) [inline Sources/AppStoreConnectClient/AppStoreConnectClient.swift]: I'm not sure that this is correct, please double check

**Review by mcmikius — COMMENTED** [review]
Need resolve comments

File: Sources/AppStoreConnectClient/Models/BuildsQuery.swift (hunk 31 above)
Comment (MykolaVasyk) [reply Sources/AppStoreConnectClient/Models/BuildsQuery.swift]: Renamed.

File: Sources/AppStoreConnectClient/Models/BuildsQuery.swift (hunk 30 above)
Comment (MykolaVasyk) [reply Sources/AppStoreConnectClient/Models/BuildsQuery.swift]: Renamed.

File: Sources/AppStoreConnectClient/AppStoreConnectClientProtocol.swift (hunk 29 above)
Comment (MykolaVasyk) [reply Sources/AppStoreConnectClient/AppStoreConnectClientProtocol.swift]: Changed.

File: Sources/AppStoreConnectClient/AppStoreConnectClient.swift (hunk 32 above)
Comment (MykolaVasyk) [reply Sources/AppStoreConnectClient/AppStoreConnectClient.swift]: Checked.

**Review by mcmikius — APPROVED** [review]

**Review by mcmikius — APPROVED** [review]

**Pull request merged**